File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/tools/utils/GuidUtils.java
package com.palitri.openiot.construction.framework.tools.utils;

import java.util.UUID;
import java.util.regex.Pattern;

public class GuidUtils {
    static public String Zero()
    {
        return "00000000-0000-0000-0000-000000000000";
    }

    static public boolean IsZero(String guid)
    {
        return GuidUtils.Zero().equals(GuidUtils.Normalize(guid));
    }

    static public boolean IsValid(String guid)
    {
        return Pattern.compile("^[0-9A-Fa-f]+$").matcher(guid).matches();
    }

    static public String FromBytes(byte[] bytes)
    {
        return GuidUtils.Normalize(StringUtils.BytesToHex(bytes));
    }

    static public byte[] ToBytes(String guid)
    {
        return StringUtils.HexToBytes(GuidUtils.Shorten(guid));
    }

    static public String Shorten(String guid)
    {
        return Pattern.compile("[^0-9A-Fa-f]").matcher(guid).replaceAll("");
    }

    static public String Normalize(String guid)
    {
        StringBuffer buffer = new StringBuffer(GuidUtils.Shorten(guid));

        buffer.insert(8, "-");
        buffer.insert(13, "-");
        buffer.insert(18, "-");
        buffer.insert(23, "-");

        return buffer.toString();
    }

    static public UUID ToGuid(String guid)
    {
        return UUID.fromString(GuidUtils.Normalize(guid));
    }

    static public String ToString(UUID guid)
    {
        return GuidUtils.Normalize(guid.toString());
    }
}

File: Windows/OpenIoT.App/OpenIoT.Lib/Tools/Utils/GuidUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Tools.Utils
{
    // Class is provided for comatibility with other system components (web api, board),
    // for fixing ambiguity between multiple existing Guid formats
    // and for serialization reasons, because OpenIoT device de/serializes guid as a stream, disregarding guid segments endianness
    public class GuidUtils
    {
        static public string Zero()
        {
            return "00000000-0000-0000-0000-000000000000";
        }

        static public bool IsZero(string guid)
        {
            return GuidUtils.Zero().Equals(GuidUtils.Normalize(guid));
        }

        static public bool IsValid(string guid)
        {
            return new Regex("^[0-9A-Fa-f]+$").IsMatch(guid);
        }

        static public string FromBytes(byte[] bytes)
        {
            return GuidUtils.Normalize(StringUtils.BytesToHex(bytes));
        }

        static public byte[] ToBytes(string guid)
        {
            return StringUtils.HexToBytes(GuidUtils.Shorten(guid));
        }

        static public string Shorten(string guid)
        {
            return new Regex("[^0-9A-Fa-f]").Replace(guid, "");
            //return guid.Replace("-", "");
        }

        static public string Normalize(string guid)
        {
            guid = GuidUtils.Shorten(guid);

            guid = guid.Insert(8, "-");
            guid = guid.Insert(13, "-");
            guid = guid.Insert(18, "-");
            guid = guid.Insert(23, "-");

            return guid;
        }

        static public Guid ToGuid(string guid)
        {
            return new Guid(GuidUtils.Normalize(guid));
        }

        static public string ToString(Guid guid)
        {
            return GuidUtils.Normalize(guid.ToString());
        }
    }
}
